### PR TITLE
Task vv/339  add popular singles episodes logic

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

File: src/screens/Home/Home.tsx [hunk 1]
```diff
@@ -20,24 +24,30 @@ import {
...
       suggestedPodcasts: home.suggestedPodcasts,
-      recentlyPlayedEpisodes: home.recentlyPlayedEpisodes
-    })
-  )
+      recentlyPlayedEpisodes: home.recentlyPlayedEpisodes,
+      popularEpisodes: home.popular
+    }))
 
   const dispatch = useDispatch()
+  const navigation = useNavigation<PodcastScreenNavigationProp>()
```
Comment (Denidzio) [inline src/screens/Home/Home.tsx]:
```suggestion
  const navigation = useNavigation<HomeScreenNavigationProp>()
```

File: src/screens/Home/Home.tsx [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+import { NavigationScreen } from 'src/common/enums'
 import { Heading, HeadingType, PlainText } from 'src/components'
 import { useAppSelector } from 'src/hooks'
 import {
   loadSuggestedPodcasts,
-  loadRecentlyPlayedEpisodes
+  loadRecentlyPlayedEpisodes,
+  loadPopularEpisodes
 } from 'src/store/actions'
+import { PodcastScreenNavigationProp } from '../Podcast/common/types'
```
Comment (Denidzio) [inline src/screens/Home/Home.tsx]:
```suggestion
import { HomeScreenNavigationProp } from './common/types'
```

File: src/screens/Home/components/PopularSingleCard/PopularSingleCard.tsx [hunk 3]
```diff
@@ -6,28 +6,39 @@ import {
...
 const PopularSingleCard: React.FC<Props> = ({
+  id,
+  position,
   title,
   author,
   source,
   style,
   onPress
 }) => {
+  const isCardRight = Boolean(position % EVEN_RATIO !== 0)
```
Comment (Denidzio) [inline src/screens/Home/components/PopularSingleCard/PopularSingleCard.tsx]: Let's avoid this calculation and do it by styles. I think `flexWrap` should help you.

**Review by Denidzio — CHANGES_REQUESTED** [review]

File: src/store/home/index.ts [hunk 4]
```diff
@@ -44,6 +51,19 @@ const homeSlice = createSlice({
     builder.addCase(addToRecentlyPlayed.fulfilled, (state, action) => {
       state.recentlyPlayedEpisodes = action.payload
     })
+    builder.addCase(loadPopularEpisodes.pending, (state) => {
+      state.dataStatus = DataStatus.PENDING
+    })
+    builder.addCase(loadPopularEpisodes.fulfilled, (state, action) => {
+      state.dataStatus = DataStatus.FULFILLED
+      state.popular = action.payload.slice(
```
Comment (eugeneprib) [inline src/store/home/index.ts]: I would change this property name from `popular` to `popularEpisodes`

**Review by eugeneprib — CHANGES_REQUESTED** [review]

**Pull request merged**